datahub-web-react/src/app/shared/entitySearch/EntitySearchResult.tsx

```diff
@@ -0,0 +1,90 @@
+import { Empty, Menu } from 'antd';
+import { MenuProps } from 'antd/lib/menu';
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import styled from 'styled-components';
+import { EntityType } from '../../../types.generated';
+import { navigateToSubviewUrl } from './routingUtils/navigateToSubviewUrl';
+import { Subview } from '../../entity/user/Subview';
+import { useEntityRegistry } from '../../useEntityRegistry';
+import EntityOwnership from './EntityOwnership';
+
+const MENU_KEY_DELIMETER = '__';
+
+const toMenuKey = (subview?: Subview, item?: string) => `${subview}${MENU_KEY_DELIMETER}${item}`;
+
+const fromMenuKey = (key: string): { subview: string; item: string } => {
+    const parts = key.split(MENU_KEY_DELIMETER);
+    return { subview: parts[0], item: parts[1] };
+};
+
+const MenuWrapper = styled.div`
+    border: 2px solid #f5f5f5;
+`;
+
+const Content = styled.div`
+    margin-left: 32px;
+    flex-grow: 1;
+`;
+
+const DetailWrapper = styled.div`
+    display: inline-flex;
+    width: 100%;
+`;
+
+type Props = {
+    urn: string;
+    ownerships: { [key in EntityType]?: any[] };
+    subview?: Subview;
+    item?: string;
+    entityType: EntityType;
+};
+
+export default function EntitySearchResult({ ownerships, subview, item, urn, entityType }: Props) {
+    const entityRegistry = useEntityRegistry();
+    const ownershipMenuOptions: Array<EntityType> = Object.keys(ownerships) as Array<EntityType>;
+    const history = useHistory();
+
+    const setSelectedEntityType = (key: string) => {
+        const { subview: nextSubview, item: nextItem } = fromMenuKey(String(key));
+        navigateToSubviewUrl({ urn, subview: nextSubview, item: nextItem, history, entityRegistry, entityType });
+    };
```

Comment:
to simplify, since we are moving to share this component, lets not handle the routing inside this component. Instead, I'd advocate for exposing an `onMenuClick` callback

Reply:
And have the callers of this component provide the logic to react.